### PR TITLE
Allow items to specify speed and sprinting restrictions while in use

### PR DESCRIPTION
**Pull request by MercurialPony**
## **Changes**
Modified _EntityPlayerSP_, added _getSpeedModifier_ and _shouldPreventSprinting_ to _Item_, added _ItemUseMovementTest_.
## **Why?**
These tweaks fix the problems with hard-coded vanilla sprinting restrictions and movement slow when using an item. It is now possible to specify exactly how much slowness will be applied and allow sprinting while the item is being used. I've seen multiple requests for this feature and I need it myself.
## **How does it work?**
It adds two new methods to the Item class allowing modders to achieve the desired effect by overriding them. Some logic has been changed in the EntityPlayerSP class to include these new methods. I've also added a test mod for anyone curious.

### CODE REVIEW

**Comment by bs2609** [comment]
Link to the previous version of this PR: #3927

File: patches/minecraft/net/minecraft/item/Item.java.patch [hunk 1]
```diff
@@ -802,6 +802,28 @@
...
 +    
++    /**
++     * Returns the movement speed multiplier which is applied to an entity using this item.
++     * @param stack the item stack being used
++     * @param entity the entity using the item
++     * @return a float between 0F and 1F
++     */
++    public float getSpeedModifier(ItemStack stack, EntityLivingBase entity)
++    {
++    	return 0.2F;
```
Comment (mezz) [inline patches/minecraft/net/minecraft/item/Item.java.patch]: please fix indentation here

File: patches/minecraft/net/minecraft/item/Item.java.patch [hunk 2]
```diff
@@ -802,6 +802,28 @@
...
++    
++    /**
++     * Whether an entity is prevented from sprinting while using this item.
++     * @param stack the item stack being used
++     * @param entity the entity using the item
++     * @return true if an entity is prevented from sprinting while using this item
++     */
++    public boolean shouldPreventSprinting(ItemStack stack, EntityLivingBase entity)
++    {
++    	return true;
```
Comment (mezz) [inline patches/minecraft/net/minecraft/item/Item.java.patch]: please fix indentation here

**Review by mezz — DISMISSED** [review]

**Review by tterrag1098 — APPROVED** [review]

**Comment by tterrag1098** [comment]
This needs an update for merge conflicts

**Comment by Melonslise** [comment]
@tterrag1098 What merge conflicts? How do I fix them?

**Comment by MercurialPony** [comment]
I lost the source and fork. Do I create a new PR?

**Comment by TechnicianLP** [comment]
try to recreate the repo/branch on the same name - if it works youre good if not you can create a new PR an link back to this one

**Pull request closed without merging**